src/openms/source/FORMAT/TseMSPFile.cpp

```diff
@@ -0,0 +1,183 @@
+// ETH Zurich, and Freie Universitaet Berlin 2002-2018.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/FORMAT/TseMSPFile.h>
+#include <OpenMS/CONCEPT/LogStream.h>
+#include <OpenMS/KERNEL/SpectrumHelper.h>
+#include <fstream>
+#include <regex>
+
+namespace OpenMS
+{
+  TseMSPFile::TseMSPFile(const String& filename, MSExperiment& experiment)
+  {
+    load(filename, experiment);
+  }
+
+  void TseMSPFile::load(const String& filename, MSExperiment& experiment) const
+  {
+    std::ifstream ifs(filename, std::ifstream::in);
+    if (!ifs.is_open())
+    {
+      throw Exception::FileNotFound(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+    }
+    const Size BUFSIZE { 65536 };
+    char line[BUFSIZE];
+    experiment.clear(true);
+    MSSpectrum spectrum;
+    bool adding_spectrum { false }; // to avoid calling `.addSpectrum()` on empty/invalid spectra
+
+    std::cmatch m;
+    std::regex re_name("^Name: (.+)");
+    std::regex re_synon("^Synon: (.+)");
+    std::regex re_formula("^Formula: (.+)");
+    std::regex re_mw("^MW: (.+)");
+    std::regex re_cas_nist("^CAS#: (.+);  NIST#: (.+)");
```

Comment:
We could search using `regex` for the "mandatory" fields as implemented, but it might be more robust for the variable fields to have some way of pulling them out based on the document structure (i.e., we know that all fields will come before the individual spectrum).  That way we will not miss any fields that we could not foresee ahead of time.

Reply:
We should also make a note in the code which fields are "mandatory" and which ones are "optional".

Reply:
Good idea. I'll try to implement that.